### PR TITLE
feat: add node input mapping with extended transformers

### PR DESCRIPTION
**Pull request by acoola**

### CODE REVIEW

**Review by olbychos — APPROVED** [review]

File: examples/dag/simple_dag.py [hunk 1]
```diff
@@ -154,6 +154,42 @@
     ).depends_on(openai_3_node)
     wf.flow.add_nodes(openai_4_node)
 
+    # OpenAI node 5 depending on OpenAI node 3 and OpenAI node 4 and used custom inputs and transformations
+    def merge_and_short_content(inputs: dict, outputs: dict[str, dict]):
```
Comment (olbychos) [inline examples/dag/simple_dag.py:158]: I can test such behaviour with more complex example in your branch?

Comment (acoola) [reply examples/dag/simple_dag.py:158]: yes

File: examples/dag/simple_dag.py [hunk 2]
```diff
@@ -154,6 +154,42 @@
     ).depends_on(openai_3_node)
     wf.flow.add_nodes(openai_4_node)
 
+    # OpenAI node 5 depending on OpenAI node 3 and OpenAI node 4 and used custom inputs and transformations
```
[the remark this answers is not in the document]
Comment (vitalii-dynamiq) [reply examples/dag/simple_dag.py:157]: @acoola

**Review by MaksymAI — APPROVED** [review]

File: dynamiq/nodes/node.py [hunk 3]
```diff
@@ -784,6 +869,65 @@ def enable_streaming(self, event: str = STREAMING_EVENT):
...
+                                    "{{extra_instructions}}\n"
+                                    "{{content}}\n"
+                                    "{{extra_content}}"
+                                ),
+                            )
+                        ],
+                    ),
+                )
+                .inputs(
+                    purpose="10 years kids",
```
Comment (vitalii-dynamiq) [inline dynamiq/nodes/node.py:916]: 10 years old kids

Comment (acoola) [reply dynamiq/nodes/node.py]: done

**Pull request merged**